File: example/mapper.js
_mkey = new org.apache.hadoop.io.Text();
_mvalue = new org.apache.hadoop.io.Text();

function map(key, value, context){
    words = value.toString().split(' ');
    for(var i=0; i < words.length; i++) {
        var word = words[i];
        _key.set(word);
        _value.set('1');
        context.write(_key,_value);
    }
}

File: phadoop-ruby/src/test/resources/mapper.rb
require 'java'
@_mkey = org.apache.hadoop.io.Text.new()
@_mvalue = org.apache.hadoop.io.Text.new()

def map(key,value,context)
    print key,value
    $_key.set(key)
    $_value.set(value)
    context.write($_key,$_value)
end
